### DISCUSSION TITLE
Can not run tests

### DISCUSSION THREAD

**Discussion by Roker2** [axmolengine/axmol#1557] [Q&A] [answered]
- axmol version: 2.1.0-rc7
- devices test on: Win10
- developing environments
   - Visual Studio: 
      - VS version: 2019 (16.11), 2022 (17.6+)
      - MSVC version: 1929, 1934, 19.35, 19.36, 19.37
      - Windows SDK version: 10.0.22621.0
Steps to Reproduce:
1. Open cpp-tests folder
2. Run `pwsh ../../build.ps1  -p win32`

It can not create `configs/config-example.plist`

**Comment by aismann** [comment]
@Roker2 
It builds using vs2022 ui?

**Reply by Roker2** [reply]
Yes

**Comment by rh101** [comment]
> 1. Open cpp-tests folder
> 
>     2. Run `pwsh ../../build.ps1  -p win32`
> 
> 
> It can not create `configs/config-example.plist`

A lot of information missing from your post.  For example, which version of Visual Studio? What is the output of the console? Did you run setup? 

A few assumptions:

1 - You've already set up as per these instructions: https://github.com/axmolengine/axmol/blob/dev/docs/DevSetup.md#prerequisites
2 - You closed the terminal and opened up a new terminal after the setup is completed

In the cpp-tests folder, run this to build:
`axmol build -p win32`

Or, if you want to create the solution/project and build it within VS, then run this:
`axmol build -p win32 -c`

**Reply by Roker2** [reply]
> Visual Studio

Visual Studio Community 2022 17.8.3

> 1 - You've already set up as per these instructions: https://github.com/axmolengine/axmol/blob/dev/docs/DevSetup.md#prerequisites
> 2 - You closed the terminal and opened up a new terminal after the setup is completed

Yes. I develop a game. But I wanted to add some code to the engine. Then... I thought I should run tests.

**Comment by Roker2** [comment]
So. I tried to use `cmake -S . -B build -G "Visual Studio 17 2022" -A x64`. I opened VS, set cpp-tests as executable target. And I got exception at `axmol\core\base\Configuration.cpp`L329

```cpp
    ValueMap dict = FileUtils::getInstance()->getValueMapFromFile(filename);
    AXASSERT(!dict.empty(), "cannot create dictionary");
```

**Reply by rh101** [reply]
That's not enough information.  Posting these issues with no useful info will unlikely lead to any helpful outcome for you.

Some examples of useful info:
- At what point of running the cpp-tests did the crash occur?
- What test were you on?
- What is in "filename" when it crashes (use the debugger and find out)?  Even better, step into the `getValueMapFromFile` method with the debugger to find out why it isn't working.
- Does the file it is referencing actually exists on the storage medium?
etc. etc.

**Reply by Roker2** [reply]
> * At what point of running the cpp-tests did the crash occur?

As I said, I set `cpp-tests` as executable target in VS. I tried to run the target `cpp-tests` and that's all. I wrote where I got error.
Also I attached link to this place in this message.
https://github.com/axmolengine/axmol/blob/3d2b6a1e07006e1df6e3d2955c3ad66808b4ace9/core/base/Configuration.cpp#L329

> * What test were you on?

How to detect it?

> * What is in "filename" when it crashes (use the debugger and find out)?

`configs/config-example.plist`

> * Does the file it is referencing actually exists on the storage medium?

I didn't create this file manually

**Reply by rh101** [reply]
> > * What is in "filename" when it crashes (use the debugger and find out)?
> 
> `configs/config-example.plist`

> > * Does the file it is referencing actually exists on the storage medium?
> 
> I didn't create this file manually

I think you misunderstood my question, as I never implied that you created that file.  My question is, did you check if the file exists? You found out the name and path of the file, being `configs/config-example.plist`, and since it's a relative path from `Contents`, then it should be `cpp-tests/Contents/configs/config-example.plist`.

Check if the file exists. If not, then something went wrong with the setup process, which @halx99 may be able to help with.  I have a feeling either the cpp-tests assets were not downloaded properly, or link was not made to them.

**Reply by Roker2** [reply]
> Check if the file exists. If not, then something went wrong with the setup process, which @halx99 may be able to help with. I have a feeling either the cpp-tests assets were not downloaded properly, or link was not made to them.

I don't see this file. And I created folder `Content` manually (I got error `No folder Content` some time ago). And I got interesting thing, that `PATH` contains `E:\Source\axmol\tools\console\bin`, but executable `axmol` is in `E:\Source\axmol\tools\console`. It was not a big problem, I added `E:\Source\axmol\tools\console` to `PATH`.
Maybe because I used old axmol version before.

**Reply by halx99** [reply]
The latest release will link `cache/sample-assets/cpp-tests/Content` to `tests/cpp-tests/Content` at cmake config step, so just delete link dest, then re-run cmake generate again.

**Reply by rh101** [reply, marked as answer]
@Roker2 You shouldn't need to edit the `PATH` env variable if you had run `setup.ps1`, nor should you be creating the `Content` folder yourself.  It may be best if you delete the Axmol folder you have at the moment, re-download version, and then run `setup.ps1`.  Everything should work after you do this.
